spring-kafka/src/main/java/org/springframework/kafka/support/converter/BytesJsonMessageConverter.java

```diff
@@ -26,14 +26,14 @@
 /**
  * JSON Message converter - {@code Bytes} on output, String, Bytes, or byte[] on input.
  * Used in conjunction with Kafka {@code BytesSerializer/BytesDeserializer}. More
- * efficient than {@link StringJsonMessageConverter} because the {@code String/byte[]} is
- * avoided.
+ * efficient than {@link StringJsonMessageConverter} because the {@code String<->byte[]}
+ * conversion is avoided.
  *
  * @author Gary Russell
  * @since 2.1.7
  *
  */
-public class BytesJsonMessageConverter extends StringJsonMessageConverter {
+public class BytesJsonMessageConverter extends JsonMessageConverter {
```

Comment:
Wait, why do we need this if we have that one generic based on the `byte[]` output?
Isn't there some automatic mechanism in Kafka to do that for us?

Reply:
I am not sure what you mean by "automatic mechanism".

I should have created the `ByteArray...` version before instead of this; creating this was a mistake, but it doesn't hurt to keep it, I think.

Reply:
I would say this my comment is a part of the `CompositeSerializer` idea. 

We configure just a `MessageConverter` and don't care what Kafka Serializer is there underneath. Just because they are all there according the classpath.

I see that there is indeed `ByteArray` and `Bytes` serializers in the Kafka, so we don't have choice and still have these two converters here any way...